### ISSUE TITLE
Why parse configuration twice?

### ISSUE THREAD

**Issue by a-sor** [dvtirol/serles-acme#4]
Hi,
Thank you for your amazing work, it has helped me a lot. I am doing a [fork](https://github.com/a-sor/serles-acme) of Serles, and I have a question. There's the following code in `serles-acme/serles/__init__.py`:

```python
def create_app():
    """ initialize web app

    This function should be passed to the WSGI server.
    """
    config, _ = get_config()

    app = Flask(__name__)
    app.config["PROPAGATE_EXCEPTIONS"] = True  # makes @app.errorhandler handle events
    app.config["SQLALCHEMY_DATABASE_URI"] = config["database"]
    app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False

    init_config()  # views.init_config()
```

Is there any special reason for parsing the config file twice, first in `config, _ = get_config()`, second in `init_config()`? I'd like to make `config` global and parse the file only once in my fork.

**Comment by a-sor** [comment]
Sorry for the late reply, lots of work. Please, take a look at how I [did](https://github.com/a-sor/serles-acme/commit/5a68718d756e38696cea492b6b0bd680b0959912#diff-617d0ad1ee32dffe865b96d25d7202d691c188613d50da68f2d8b07d21720350) it. Yes, this most likely breaks unit tests, but this way we a) avoid excessive parsing of configuration and b) gain access to configuration globally using `serles.config[...]`.

I am interested in some new features for Serles, which I am working on, but I would also like to keep in sync with you somehow. Anyway, I would appreciate any comments from you.

**Comment by uedvt359** [comment]
i'm all for improving this, but breaking unit tests is unacceptable. i too think it's good to stay close to each other (code-wise). can you please submit a pull request with your proposed config initialization changes? the commit you posted in your last comment does a lot of other stuff, which makes it hard to follow.

**Comment by a-sor** [comment]
Hi, I've submitted a pull request. Without my other changes, unrelated to the variables `config` and `backend`, the patch turned out to be very short. And, surprisingly, it doesn't seem to break unit tests! At least, I managed to run successfully those in the tests directory:
```
$ make -f tests/Makefile
PYTHONPATH=.:./lib coverage3 run -m pytest tests
========================================================================== test session starts ==========================================================================
platform linux -- Python 3.10.12, pytest-8.1.1, pluggy-1.4.0
rootdir: /tmp/serles-acme-patched
collected 77 items

tests/test_appfactory.py .                                                                                                                                        [  1%]
tests/test_certbotbackend.py .......                                                                                                                              [ 10%]
tests/test_challengefuns.py .......................                                                                                                               [ 40%]
tests/test_configparser.py ....                                                                                                                                   [ 45%]
tests/test_ejbcabackend.py .....                                                                                                                                  [ 51%]
tests/test_flaskfuns.py ................                                                                                                                          [ 72%]
tests/test_helpers.py ........                                                                                                                                    [ 83%]
tests/test_opensslbackend.py ..                                                                                                                                   [ 85%]
tests/test_views.py ...........                                                                                                                                   [100%]

=========================================================================== warnings summary ============================================================================
tests/test_certbotbackend.py::CertbotBackendTester::test_backend_okconfig
  /tmp/serles-acme-patched/tests/test_certbotbackend.py:45: DeprecationWarning: Please use assertNotEqual instead.
    self.assertNotEquals(backend, None)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
===================================================================== 77 passed, 1 warning in 2.17s =====================================================================
coverage3 html --omit=serles/__main__.py,serles/backends/base.py serles/*.py serles/backends/*.py
Wrote HTML report to htmlcov/index.html

```
